pytest_splunk_addon/standard_lib/utilities/sample_splitter.py

```diff
@@ -0,0 +1,91 @@
+#
+# Copyright 2021 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import xmltodict
+import os
+import argparse
+
+
+def split_samples(sample_path, output_dir=None, splitter="sourcetype"):
+    """
+    Splits events to separate files based on provided field
+
+    Args:
+        sample_path:  path to file with samples
+        output_dir:   path to directory where separated files will be stored. Default the same dir as input file
+        splitter:     depend on what field samples are split. Default sourcetype, allowed source and host
+    """
+    if output_dir is None:
+        output_dir = os.path.dirname(sample_path)
+    with open(sample_path, "r", encoding="utf-8") as sample_file:
+        sample_raw = sample_file.read()
+    samples = xmltodict.parse(sample_raw)
+    events = (
+        samples["device"]["event"]
+        if type(samples["device"]["event"]) == list
+        else [samples["device"]["event"]]
+    )
+    separate_events = {}
+    for each_event in events:
+        if "transport" in each_event.keys() and each_event["transport"].get(
+            f"@{splitter}"
+        ):
+            splitter_name = each_event["transport"].get(f"@{splitter}")
+            if splitter_name in separate_events.keys():
```

Comment:
If you define `seperate_events` as `seperate_events = defaultdict(list)` you can avoid checking if `splitter_name` is already in dict